src/renderer/api/terminal-api.ts

```diff
@@ -55,42 +57,43 @@ export class TerminalApi extends WebSocketApi {
   protected size: { Width: number; Height: number };
 
   public onReady = new EventEmitter<[]>();
-  public isReady = false;
+  @observable public isReady = false;
+  @observable public rcFinished = false;
+  public readonly url: string;
 
   constructor(protected options: TerminalApiQuery) {
     super({
       logging: isDevelopment,
       flushOnOpen: false,
       pingIntervalSeconds: 30,
     });
-  }
-
-  async getUrl() {
-    let { port } = location;
-    const { hostname, protocol } = location;
-    const { id, node } = this.options;
-    const wss = `ws${protocol === "https:" ? "s" : ""}://`;
-    const query: TerminalApiQuery = { id };
+    makeObservable(this);
 
-    if (port) {
-      port = `:${port}`;
-    }
+    const { hostname, protocol, port } = location;
+    const query: ParsedUrlQueryInput = {
+      id: options.id,
+    };
 
-    if (node) {
-      query.node = node;
-      query.type = "node";
+    if (options.node) {
+      query.node = options.node;
+      query.type = options.type || "node";
     }
 
-    return `${wss}${hostname}${port}/api?${stringify(query)}`;
+    this.url = url.format({
+      protocol: protocol.includes("https") ? "wss" : "ws",
+      hostname,
+      port,
+      pathname: "/api",
+      query,
+      slashes: true,
+    });
```

Comment:
This was done because `getUrl()` both doesn't need to be `async` and because it should never change, so we might as well only do the work once.